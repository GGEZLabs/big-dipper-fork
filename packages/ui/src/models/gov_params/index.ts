import * as R from 'ramda';

class GovParams {
  public params: {
    depositParams: {
      minDeposit: {
        denom: string;
        amount: string;
      }[];
      maxDepositPeriod: number;
    };
    tallyParams: {
      quorum: string;
      threshold: string;
      vetoThreshold: string;
    };
    votingParams: {
      votingPeriod: number;
    };
  };

  constructor(payload: any) {
    this.params = payload.params;
  }

  static fromJson(data: any) {
    return new GovParams({
      params: {
        depositParams: {
          minDeposit: R.pathOr([], ['params', 'min_deposit'], data).map((x) => ({
            denom: x.denom,
            amount: x.amount,
          })),
          maxDepositPeriod: R.pathOr(0, ['params', 'max_deposit_period'], data),
        },
        tallyParams: {
          quorum: R.pathOr('0', ['params', 'quorum'], data),
          threshold: R.pathOr('0', ['params', 'threshold'], data),
          vetoThreshold: R.pathOr('0', ['params', 'veto_threshold'], data),
        },
        votingParams: {
          votingPeriod: R.pathOr(0, ['params', 'voting_period'], data),
        },
      },
    });
  }
}

export default GovParams;
