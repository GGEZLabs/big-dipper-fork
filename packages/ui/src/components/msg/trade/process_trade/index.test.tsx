import renderer from 'react-test-renderer';
import ProcessTrade from '@/components/msg/trade/process_trade';
import { MsgProcessTrade } from '@/models';
import MockTheme from '@/tests/mocks/MockTheme';

// ==================================
// mocks
// ==================================
jest.mock('@/components/name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Name" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/MsgProcessTrade', () => {
  it('matches snapshot', () => {
    const message = MsgProcessTrade.fromJson({
      category: 'trade',
      type: 'MsgProcessTrade',
      creator: 'GGEZ1 Foundation',
    });
    const component = renderer.create(
      <MockTheme>
        <ProcessTrade message={message} />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
