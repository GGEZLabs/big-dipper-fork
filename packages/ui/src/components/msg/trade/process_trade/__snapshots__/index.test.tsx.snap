// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/MsgProcessTrade matches snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Hind Madurai",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<p
  className="MuiTypography-root MuiTypography-body1 emotion-0"
>
  <span
    data-testid="Trans"
    i18nKey="message_contents:txProcessTradeContent"
  >
    <div
      address="GGEZ1 Foundation"
      id="Name"
      name="GGEZ1 Foundation"
    />
  </span>
</p>
`;
